### PR TITLE
[ES-14934] Retrying!

### PR DESCRIPTION
**Pull request by databricks-david-lewis**
Retry failed chunks!

### CODE REVIEW

File: databricks_cli/dbfs/api.py [hunk 1]
```diff
@@ -115,13 +116,23 @@ def get_file(self, dbfs_path, dst_path, overwrite, headers=None):
...
-                local_file.write(b64decode(data))
+                try:
+                    response = self.client.read(dbfs_path.absolute_path, offset, BUFFER_SIZE_BYTES,
+                                                headers=headers)
+                    bytes_read = response['bytes_read']
+                    data = response['data']
+                    offset += bytes_read
+                    local_file.write(b64decode(data))
+                    retries_left = 5
+                except:
```
Comment (mparkhe) [inline databricks_cli/dbfs/api.py]: except as e:

File: databricks_cli/dbfs/api.py [hunk 2]
```diff
@@ -115,13 +116,23 @@ def get_file(self, dbfs_path, dst_path, overwrite, headers=None):
...
+                    if retries_left > 0:
+                        time.sleep(1)
+                        print 'Error reading chunk, trying again (', retries_left, ' left)!'
```
Comment (mparkhe) [inline databricks_cli/dbfs/api.py]: if you want to print exception ...
`print(str(e) + 'Error reading ... ')`

File: databricks_cli/dbfs/api.py [hunk 3]
```diff
@@ -115,13 +116,23 @@ def get_file(self, dbfs_path, dst_path, overwrite, headers=None):
...
+                        retries_left = retries_left - 1
+                    else:
+                        raise
```
Comment (mparkhe) [inline databricks_cli/dbfs/api.py]: `raise e`

File: databricks_cli/dbfs/api.py [hunk 4]
```diff
@@ -115,13 +116,23 @@ def get_file(self, dbfs_path, dst_path, overwrite, headers=None):
...
+                try:
+                    response = self.client.read(dbfs_path.absolute_path, offset, BUFFER_SIZE_BYTES,
+                                                headers=headers)
+                    bytes_read = response['bytes_read']
+                    data = response['data']
+                    offset += bytes_read
+                    local_file.write(b64decode(data))
+                    retries_left = 5
+                except:
+                    if retries_left > 0:
```
Comment (andrewmchen) [inline databricks_cli/dbfs/api.py]: I think we should add a unit test for this addition piece of logic.

Comment (andrewmchen) [reply databricks_cli/dbfs/api.py]: Something like this should do the trick:
```

    def test_get_file_with_error(self, dbfs_api, tmpdir):
        api_mock = dbfs_api.client
        api_mock.get_status.return_value = TEST_FILE_JSON
        api_mock.read.side_effect = [Exception(), {
            'bytes_read': 1,
            'data': b64encode(b'x'),
        }]

        test_file_path = os.path.join(tmpdir.strpath, 'test')
        dbfs_api.get_file(TEST_DBFS_PATH, test_file_path, True)

        with open(test_file_path, 'r') as f:
            assert f.read() == 'x'
```

**Pull request closed without merging**